File: Memory.py
from Hardware import Hardware

class Memory(Hardware):

    def __init__(self, id, p, b, m, f, ms, sc):
        super().__init__(id, p, b, m)
        self.frequency = f
        self.memorySize = ms
        self.supportsCuda = sc

    def setFrequency(self, f):
        self.frequency = f
    def getFrequency(self):
        return self.frequency

    def setMemorySize(self, ms):
        self.memorySize = ms
    def getMemorySize(self):
        return self.memorySize

    def setSupportsCuda(self, sc):
        self.supportsCuda = sc
    def getSupportsCuda(self):
        return self.supportsCuda

File: CPP/Memory.cpp
#include <iostream>
#include <string>

class Memory : public Hardware{
    private:
        string frequency;
        string memorySize;
        bool supportsCuda;

    public:
        Memory(){
        }

        Memory(string f, string m, bool s){
            this->frequency = f;
            this->memorySize = m;
            this->supportsCuda = s;
        }

        void setFrequency(string f){
            this->frequency = f;
        }
        string getFrequency(){
            return this->frequency;
        }

        void setMemorySize(string m){
            this->memorySize = m;
        }
        string getMemorySize(){
            return this->memorySize;
        }

        void setSupportsCuda(bool f){
            this->supportsCuda = f;
        }
        bool getSupportsCuda(){
            return this->supportsCuda;
        }

        ~Memory(){
        }
};